android/game/src/main/java/com/farsitel/bazaar/game/GameHubBroadcast.java

```diff
@@ -0,0 +1,130 @@
+                    return;
+                }
+
+                if (callbacks.containsKey(action)) {
+                    Result result = new Result();
+                    if (action.equals(IS_LOGIN)) {
+                        boolean isLogin = intent.getBooleanExtra(ParamNames.IS_LOGIN, false);
+                        result.status = isLogin ? Status.SUCCESS : Status.LOGIN_CAFEBAZAAR;
+                        result.message = isLogin ? "" : "Login to Cafebazaar before!";
+                        callbacks.get(action).call(result);
+                    } else {
+                        callbacks.get(action).call(result.setBundle(intent.getExtras()));
+                    }
+                    callbacks.remove(action);
+                }
+            }
+        };
+    }
+
+    private void registerBroadcast(Context context) {
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(IS_LOGIN);
+        intentFilter.addAction(GET_TOURNAMENTS);
+        intentFilter.addAction(START_TOURNAMENT_MATCH);
+        intentFilter.addAction(END_TOURNAMENT_MATCH);
+        intentFilter.addAction(GET_CURRENT_LEADERBOARD_DATA);
+        context.registerReceiver(receiver, intentFilter);
+    }
+
+    private void sendBroadcast(String action, Bundle extras) {
+        extras.putString(ParamNames.PACKAGE_NAME, context.getPackageName());
+
+        Intent broadcastIntent = new Intent();
+        broadcastIntent.setPackage(AbstractGameHub.BAZAAR_PACKAGE_NAME);
+        broadcastIntent.putExtras(extras);
+        broadcastIntent.setAction(action);
+        context.sendBroadcast(broadcastIntent);
+    }
+
+    public void isLogin(IBroadcastCallback callback) {
+        callbacks.put(IS_LOGIN, callback);
+        sendBroadcast(IS_LOGIN, new Bundle());
+    }
+
+    public void getTournamentTimes(IBroadcastCallback callback) {
+        callbacks.put(GET_TOURNAMENTS, callback);
+        sendBroadcast(GET_TOURNAMENTS, new Bundle());
+    }
+
+    public void startTournamentMatch(String matchId, String metadata, IBroadcastCallback callback) {
+        Bundle extras = new Bundle();
+        extras.putString(ParamNames.MATCH_ID, matchId);
+        extras.putString(ParamNames.META_DATA, metadata);
+        callbacks.put(START_TOURNAMENT_MATCH, callback);
+        sendBroadcast(START_TOURNAMENT_MATCH, extras);
+    }
+
+    public void endTournamentMatch(String sessionId, float score, IBroadcastCallback callback) {
+        Bundle extras = new Bundle();
+        extras.putString(ParamNames.SESSION_ID, sessionId);
+        extras.putFloat(ParamNames.SCORE, score);
+        callbacks.put(END_TOURNAMENT_MATCH, callback);
+        sendBroadcast(END_TOURNAMENT_MATCH, extras);
+    }
+
+    public void getCurrentLeaderboard(IBroadcastCallback callback) {
+        callbacks.put(GET_CURRENT_LEADERBOARD_DATA, callback);
+        sendBroadcast(GET_CURRENT_LEADERBOARD_DATA, new Bundle());
```

Comment:
Why bundle is empty?

Reply:
It fills in method
I overload sendBroadcast method